I'm going to retrieve ERA5 data (not the entire 270GB), only partly via cdsapi, I used google colab to try this notebook

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Retrieve ERA5 data via cdsapi

In [2]:
!pip install cdsapi


In [3]:
import cdsapi
client = cdsapi.Client(url='https://cds.climate.copernicus.eu/api', key='e47812d5-3792-40c1-9ead-261a90679779')


In [4]:
import os

project_path = '/content/drive/MyDrive/WindNowCasting'
os.chdir(project_path)

print(f"Current working directory changed to: {os.getcwd()}")

Current working directory changed to: /content/drive/MyDrive/WindNowCasting


Try to retrieve 1 hour data


In [5]:
dataset = 'reanalysis-era5-single-levels'
request = {
    'product_type': 'reanalysis',
    'variable': ['10m_u_component_of_wind', '10m_v_component_of_wind'],
    'year': ['2024'],
    'month': ['03'],
    'day': ['01'],
    'time':['13:00'],
    'data_format': 'netcdf',
}
target = 'wind_2024_03_01_13.nc'

client.retrieve(dataset, request, target)

2026-04-20 03:56:47,078 INFO [2025-12-11T00:00:00] Please note that a dedicated catalogue entry for this dataset, post-processed and stored in Analysis Ready Cloud Optimized (ARCO) format (Zarr), is available for optimised time-series retrievals (i.e. for retrieving data from selected variables for a single point over an extended period of time in an efficient way). You can discover it [here](https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-timeseries?tab=overview)
INFO:ecmwf.datastores.legacy_client:[2025-12-11T00:00:00] Please note that a dedicated catalogue entry for this dataset, post-processed and stored in Analysis Ready Cloud Optimized (ARCO) format (Zarr), is available for optimised time-series retrievals (i.e. for retrieving data from selected variables for a single point over an extended period of time in an efficient way). You can discover it [here](https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-timeseries?tab=overview)
2026-

afbd7ad61e4c7ffbd4159b58ef14a957.nc:   0%|          | 0.00/3.79M [00:00<?, ?B/s]

'wind_2024_03_01_13.nc'

Retrieve 1 month data (February 2024), all the hourly data

In [6]:
client.retrieve(
    'reanalysis-era5-single-levels',
    {
        'product_type': 'reanalysis',
        'variable': [
            '10m_u_component_of_wind',
            '10m_v_component_of_wind',
            '2m_temperature'
        ],
        'year': '2024',
        'month': '02',
        'day': [f'{d:02d}' for d in range(1, 32)],
        'time': [f'{h:02d}:00' for h in range(24)],  # all 24 hours
        'format': 'netcdf',
    },
    'era5_Feb2024.nc'
)

2026-04-20 03:58:22,512 INFO [2025-12-11T00:00:00] Please note that a dedicated catalogue entry for this dataset, post-processed and stored in Analysis Ready Cloud Optimized (ARCO) format (Zarr), is available for optimised time-series retrievals (i.e. for retrieving data from selected variables for a single point over an extended period of time in an efficient way). You can discover it [here](https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-timeseries?tab=overview)
INFO:ecmwf.datastores.legacy_client:[2025-12-11T00:00:00] Please note that a dedicated catalogue entry for this dataset, post-processed and stored in Analysis Ready Cloud Optimized (ARCO) format (Zarr), is available for optimised time-series retrievals (i.e. for retrieving data from selected variables for a single point over an extended period of time in an efficient way). You can discover it [here](https://cds.climate.copernicus.eu/datasets/reanalysis-era5-single-levels-timeseries?tab=overview)
2026-

279381a21bbe98bbd878914c5aaf57e7.nc:   0%|          | 0.00/3.62G [00:00<?, ?B/s]

'era5_Feb2024.nc'

total data size of 1 month data hourly : 3.6 GB